src/paseto_v2_public.c

```diff
@@ -21,46 +21,54 @@ static const size_t signature_len = crypto_sign_BYTES;
 
 
 bool paseto_v2_public_load_public_key_hex(
-        v2_public_pk key,
+        struct paseto_v2_public_pk *key,
         const char *key_hex) {
-        key.header = V2_PUBLIC;
-    return key_load_hex(key.key_bytes, paseto_v2_PUBLIC_PUBLICKEYBYTES, key_hex);
+        struct paseto_v2_public_pk tmp;
+        *key = &tmp;
+        key->header = V2_PUBLIC;
+    return key_load_hex(key->key_bytes, paseto_v2_PUBLIC_PUBLICKEYBYTES, key_hex);
 }
 
 
 bool paseto_v2_public_load_public_key_base64(
-        v2_public_pk key,
+        struct paseto_v2_public_pk *key,
         const char *key_base64) {
-        key.header = V2_PUBLIC;
-    return key_load_base64(key.key_bytes, paseto_v2_PUBLIC_PUBLICKEYBYTES, key_base64);
+        struct paseto_v2_public_pk tmp;
+        *key = &tmp;
+        key->header = V2_PUBLIC;
+    return key_load_base64(key->key_bytes, paseto_v2_PUBLIC_PUBLICKEYBYTES, key_base64);
 }
 
 
 bool paseto_v2_public_load_secret_key_hex(
-        v2_public_sk key,
+        struct paseto_v2_public_sk key,
         const char *key_hex) {
-        key.header = V2_PUBLIC;
-    return key_load_hex(key.key_bytes, paseto_v2_PUBLIC_SECRETKEYBYTES, key_hex);
+        struct paseto_v2_public_sk tmp;
+        *key = &tmp;
+        key->header = V2_PUBLIC;
+    return key_load_hex(key->key_bytes, paseto_v2_PUBLIC_SECRETKEYBYTES, key_hex);
 }
 
 
 bool paseto_v2_public_load_secret_key_base64(
         v2_public_sk key,
         const char *key_base64) {
-        key.header = V2_PUBLIC;
-    return key_load_base64(key.key_bytes, paseto_v2_PUBLIC_SECRETKEYBYTES, key_base64);
+        struct paseto_v2_public_sk tmp;
+        *key = &tmp;
```

Comment:
There's no point in doing this, since tmp does not get initialized in C, you can just drop it. Also the `&tmp` is plain wrong, you can't the address of a variable on the stack and returns it; aside from the fact that this doesn't even compile. Please test your patches before you send them, the readme contains the few simple instructions necessary.